### ISSUE TITLE
Can't kill Docker container with `ctrl-c`

### ISSUE THREAD

**Issue by altano** [zerotier/ZeroTierOne#1580]
I'm trying to run a docker container in my terminal using [the command given on Docker Hub](https://hub.docker.com/r/zerotier/zerotier):

```
docker run --name myzerotier --rm --cap-add NET_ADMIN --device /dev/net/tun zerotier/zerotier:1.8.4 <REDACTED_NETWORK_ID>
```

When I do this, `ctrl-c` does nothing in my terminal. The only way to kill the container is to use `docker kill <container>`.

I'm on Windows 11 using Windows Terminal and I also tried with the VS Code JS terminal. I'm SSH'd into my Docker host (an Ubuntu 20.04 LXC and running the command there).

I took a look at [entrypoint.sh.release](https://github.com/zerotier/ZeroTierOne/blob/master/entrypoint.sh.release) and I couldn't figure out the issue.

**Comment by erikh** [comment]
I'll try to look at this one tomorrow.

------- Original Message -------

**Comment by glimberg** [comment]
`docker run` traps Ctrl-C.  If you add `-i -t` to your command line, becoming:
```
docker run --name myzerotier --rm --cap-add NET_ADMIN --device /dev/net/tun -i -t zerotier/zerotier:1.8.4 <REDACTED_NETWORK_ID>
```
Ctrl-C will work as expected.  As a side effect, it also solves your other issue https://github.com/zerotier/ZeroTierOne/issues/1581 since ZeroTier gets the ctrl-c signal, shuts down properly and removes its own .pid file.

**Comment by erikh** [comment]
This is interesting because we specifically trap `SIGINT` and `SIGTERM` in the shell script, and _explicitly kill zerotier before exiting the container_. I don't think this is us, but if it is, the changes I filed in #1596 should resolve it.

**Comment by erikh** [comment]
I did test this earlier, too. With and without `-i -t` flags the container exits cleanly. Could be a windows terminal issue.

**Comment by glimberg** [comment]
I can't speak for windows.  I can CTRL-C definitely isn't working to stop the container without the `-i -t` flags on macOS & Linux

**Comment by someara** [comment]
This one rings a bell...
Can you test this pull request and see if it works for you?

https://github.com/zerotier/ZeroTierOne/pull/1597

**Comment by glimberg** [comment]
Checked out the latest changes in #1596.  Still no joy when hitting `ctrl-c` without `-it`.  `docker stop ` from another terminal works fine, but `ctrl-c` does nothing

```
docker run --name myzerotier --rm --cap-add NET_ADMIN --device /dev/net/tun -v /tmp/zerotier-db:/var/lib/zerotier-one zerotier/zerotier:1.8.6 xxxxxxxxxxd49dab
starting zerotier
zerotier hasn't started, waiting a second
tail: cannot open 'nohup.out' for reading: No such file or directory
joining networks: xxxxxxxxxxd49dab
joining xxxxxxxxxxd49dab
200 join OK
^C^C^C
```

**Comment by glimberg** [comment]
Still no `CTRL-C` in your PR @someara

**Comment by erikh** [comment]
closed by #1596
